backend/src/connectors/store.py

```
@@ -61,35 +76,48 @@ def iterate_qdrant_docs(
-    files = source.list_files()
-
+def build_vectorstore(files: List[VDBFile], source: str, batch_size=200):
     # Read status manifest file
     manifest = VDBManifest(QDRANT_PATH)
+    current_config_hash = get_config_hash()
+
+    config_changed = False
+    if hasattr(manifest, "needs_config_rebuild"):
+        try:
+            config_changed = bool(manifest.needs_config_rebuild(current_config_hash))
+        except Exception:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)